File: 5052.cc
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool isConsistence = 1;

class Trie {
	struct vector<Trie*> node;
	int key;
	bool isMatched;

public:
	Trie(int key) {
		this->key = key;
		this->isMatched = 0;
		this->node = vector<Trie*>(10, NULL);
	}
	
	void find(string &str, int index) {
		if (str.size() == index) {
			return;
		}

		int cur = str[index] - '0';
		if (this->node[cur] == NULL) {
			return;
		}

		if (str.size() > index && this->isMatched && this->key != -1) {
			isConsistence = 0;
		}

		this->node[cur]->find(str, index + 1);
	}

	void insert(string &str, int index) {
		int cur = str[index] - '0';
		if (this->node[cur] == NULL) {
			this->node[cur] = new Trie(cur);
		}
		if (str.size() == index + 1) {
			this->node[cur]->isMatched = 1;
			return;
		}
		this->node[cur]->insert(str, index + 1);
	}
};

int main() {
	int t, n;
	cin >> t;
	while (t--) {
		isConsistence = 1;
		Trie* trie = new Trie(-1);
		cin >> n;
		vector<string> v(n);
		for (int i = 0; i < n; ++i) {
			cin >> v[i];
			trie->insert(v[i], 0);
		}
		sort(v.begin(), v.end());
		for (string& s : v) {
			trie->find(s, 0);
		}
		delete trie;
		cout << (isConsistence ? "YES" : "NO") << '\n';
	}
}

File: 5052.py
is_consistence = True

class Trie:
    def __init__(self, key):
        self.node = {}
        self.count = {}
        self.key = key
        self.is_matched = False

    def find(self, key):
        if key != '' and self.is_matched:
            global is_consistence
            is_consistence = False
            return

        if self.node.get(key[:1]) is None:
            return None
        else:
            return self.node[key[:1]].find(key[1:])

    def insert(self, key):
        if key == '':
            self.is_matched = True
            return

        if self.node.get(key[:1]) is None:
            self.node[key[:1]] = Trie(key[:1])
        self.node[key[:1]].insert(key[1:])


t = int(input())

for _ in range(t):
    is_consistence = True
    arr = []
    root = Trie(None)
    n = int(input())
    for i in range(n):
        arr.append(input())
        root.insert(arr[-1])
    arr.sort()
    for i in arr:
        root.find(i)
    print('YES' if is_consistence else 'NO')